sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -992,9 +998,35 @@ class CudfRegexTranspiler(mode: RegexMode) {
 
     checkUnsupported(regex)
 
+    var current = 0
+    // capture groups can be nested, so we need to do this logic outside of the rewrite
+    def updateGroupsForExtract(regex: RegexAST, n: Int): RegexAST = {
+      regex match {
+        case RegexGroup(capture, term, lookahead) if capture => {
+          current += 1
+          if (n != current) {
+            RegexGroup(false, updateGroupsForExtract(term, n), lookahead)
+          } else {
+            RegexGroup(true, updateGroupsForExtract(term, n), lookahead)
+          }
```

Comment:
It looks like this could be simplified.

```suggestion
          RegexGroup(n == current, updateGroupsForExtract(term, n), lookahead)
```